frontend/src/framework/internal/components/PersistenceManagementDialog/sessionManagementContent.tsx

```diff
@@ -0,0 +1,419 @@
+        },
+    },
+
+    // TODO - Future work: Could be nice to show/filter on modules used, but need backend changes and virtual table columns to support that
+    // {
+    //     _type: "virtual",
+    //     columnId: "modules" as keyof SessionMetadataWithId_api,
+    //     label: "Modules",
+    //     sizeInPercent: 15,
+    //     filter: false,
+    // },
+    {
+        _type: "data",
+        columnId: "updatedAt",
+        label: "Updated at",
+        sizeInPercent: 15,
+        filter: false,
+        formatValue: (value) => {
+            return formatDate(new Date(value));
+        },
+    },
+    {
+        _type: "data",
+        columnId: "createdAt",
+        label: "Created at",
+        sizeInPercent: 15,
+        filter: false,
+        formatValue: (value) => {
+            return formatDate(new Date(value));
+        },
+    },
+];
+
+function columnIdToApiSortField(columnId: string): SessionSortBy_api {
+    switch (columnId) {
+        case "title":
+            return SessionSortBy_api.METADATA_TITLE;
+        case "updatedAt":
+            return SessionSortBy_api.METADATA_UPDATED_AT;
+        case "createdAt":
+            return SessionSortBy_api.METADATA_CREATED_AT;
+
+        default:
+            throw new Error(`Unknown columnId: ${columnId}`);
+    }
+}
+
+function tableSortDirToApiSortDir(sort: TableSortDirection): SortDirection_api {
+    return sort as unknown as SortDirection_api;
+}
+
+function useInfiniteSessionMetadataQuery(
+    querySortParams: Options<GetSessionsMetadataData_api>["query"],
+    enabled?: boolean,
+) {
+    // ! We need to manually write out the query because hey-api generates keys in a way that messes with Tanstack's
+    // ! ability to set query data (which we use after mutating metadata).
+    // ! You'd think this would work, but if I try this; the data never loads, because it tries to get the query
+    // ! params from the key...
+    // return useInfiniteQuery({
+    //     ...getSessionsMetadataInfiniteOptions({
+    //         query: {
+    //             ...querySortParams,
+    //             limit: QUERY_PAGE_SIZE,
+    //             // // TODO: Rename `cursor` to `continuation_token` once we update to latest hey-api version
+    //             // cursor: pageParam,
+    //         },
+    //     }),
+    //     queryKey: [
+    //         // @ts-expect-error -- Ignore expected tanstack key type
+    //         "getSessionsMetadata",
+    //         "infinite",
+    //         querySortParams?.filter_title,
+    //         querySortParams?.filter_updated_from,
+    //         querySortParams?.filter_updated_to,
+    //         querySortParams?.sort_by,
+    //         querySortParams?.sort_direction,
+    //     ],
+    //     initialPageParam: null,
+    //     refetchInterval: 20000,
+    //     getNextPageParam(lastPage) {
+    //         return lastPage.continuation_token;
+    //     },
+    // });
```

Comment:
Should we make the comment more compact? It is informative, can keep as is if we feel the need.